internal/datastore/postgres/revisions_test.go

```diff
@@ -73,9 +74,103 @@ func TestRevisionSerDe(t *testing.T) {
 			serialized := rev.String()
 			require.Equal(tc.expectedStr, serialized)
 
-			parsed, err := parseRevision(serialized)
+			parsed, err := parseRevisionProto(serialized)
 			require.NoError(err)
 			require.Equal(rev, parsed)
 		})
 	}
 }
+
+func TestRevisionBackCompat(t *testing.T) {
+	testCases := []struct {
+		snapshot  pgSnapshot
+		inputStr  string
+		expectErr bool
+	}{
+		{snap(1, 1), "0", false},
+		{snap(1, 1), "0.0", false},
+		{snap(501, 501), "500", false},
+		{snap(499, 501, 499), "500.499", false},
+		{snap(499, 507, 499, 500, 501, 502, 503, 504, 505), "506.499", false},
+		{snap(maxInt+1, maxInt+1), "9223372036854775807", false},
+		{snap(maxInt-1, maxInt+1, maxInt-1), "9223372036854775807.9223372036854775806", false},
+		{snap(0, 0), "-500", true},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.snapshot.String(), func(t *testing.T) {
+			require := require.New(t)
+
+			parsed, err := parseRevisionDecimal(tc.inputStr)
```

Comment:
shouldn't this be testing the parseRevision overall?

edit: rename the test to TestRevisionParseOldRevisionFormat or somesuch